apis/ec2/v1alpha1/custom_types.go

```diff
@@ -32,3 +32,54 @@ type CustomVPCPeeringConnectionParameters struct {
 	// connection if both VPCs are in the same tenant.
 	AcceptRequest bool `json:"acceptRequest,omitempty"`
 }
+
+// CustomTransitGatewayParameters are custom parameters for TransitGateway
+type CustomTransitGatewayParameters struct{}
+
+// CustomTransitGatewayVPCAttachmentParameters are custom parameters for TransitGatewayVPCAttachment
+type CustomTransitGatewayVPCAttachmentParameters struct {
+	// The ID of the VPC.
+	// +optional
+	VPCID *string `json:"vpcID,omitempty"`
```

Comment:
Nitpick: while we don't have explicit convention around this, I think it's fair to go with calculated lower camel, i.e. `vpcId`, instead of taking known words into account. There are some places we do one and other places we do the other, but I believe calculated lower camel is the most consistent for repos where there is code generation, like this one or Jet-based providers.